---
title: "Introduction to biomestats"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to biomestats}
  %\VignetteEngine{knitr::rmarkdown}
  \usepackage[utf8]{inputenc}
---
  
```{r setup, include = FALSE}
knitr::opts_chunk$set(
collapse = TRUE,
comment = "#>"
)
```

# Import libraries  

```{r eval=TRUE, message=FALSE, warning=FALSE}
library(biomestats)
```


```{r eval=TRUE}
dis_col <- c(acute_ili="#6883ba", no_ili = "#166f65", `NA`="black") 
```

# Import data  
```{r eval=TRUE}
data("ili_data")
ps <- ili_data

table(meta(ps)$condition_status)
#taxa_names(ps)
taxa_names(ps) <- gsub(" ", ".", taxa_names(ps))
taxa_names(ps) <- gsub("\\[|\\]", "", taxa_names(ps))
taxa_names(ps) <- gsub("-", ".", taxa_names(ps))

sample_data(ps)[sample_data(ps)[, c(colnames(sample_data(ps)))] == "unknown"] <- 2
```

```{r}
# Recode smoking variables: so 0 is no and increasing to yes 2
sample_data(ps)$Roken_2014 <- 3-as.numeric(sample_data(ps)$Roken_2014)
```


## Abundance data  

Prepare a tibble from phyloseq object. This tibble consists of ID (for sample) relative abundance of taxa, average and SD of counts of taxa in sample, prop.positive, entropy and C.V.                      

```{r eval=TRUE}

gen_tab <- transform_to_fractions(ps, 
                                  det.thres= 0.001, 
                                  prev.thres=10/100)

dim(gen_tab)
```

## Variables to test   

There can be numberous variables that a user might want to test. It is advised to select and specify variables of interest. This has to be done manually because sample data my consists of several other data that are not useful, like primer/barcode sequences, internal codes from sequencing center, etc.  

```{r eval=TRUE}

#main_var <- "ILI"

comord.vars <- c("BMI_2014","chronic_cvd_2014", "Roken_2014","age_yrs_oct2014")

med.vars <- c("antimicrobials_2014", "AstmaCOPD_2014", "statins_2014", "alpa_beta_blockers_2014")

all.vars <- c(comord.vars,med.vars)
```

```{r eval=TRUE}


meta_dat <- meta(ps)
table(meta_dat$condition_status,meta_dat$antimicrobials_2014, useNA = "always")

table(meta_dat$condition_status,meta_dat$respiratory_disease_2014, useNA = "always")
```

need to *test* 

## Prep metadata  

```{r eval=TRUE}
phenotypic.data.1 <- prep_metadata(ps, 
                                   sam.ids = "seq_sam_name", 
                                   ref.group="no_ili", 
                                   compare= "condition_status", 
                                   confounder.vars = c("gender", "age_yrs_oct2014"),
                                   var.interest=all.vars)

head(phenotypic.data.1)

```
## Strata frequency  
```{r eval=TRUE}

frequency.of.strata <- table(phenotypic.data.1$stratum)
#mean(frequency.of.strata)
frequency.of.strata

```


## Data for testing
```{r eval=TRUE}


taxa_select <- colnames(gen_tab)[2:(ncol(gen_tab)-5)]

other <- c("average","SD","prop.positive","entropy","CV")

#vars.test <- colnames(phenotypic.data.1)[c(2,5:39)]
vars.test <- c("ILI",all.vars)

data.set.1 <- phenotypic.data.1 %>% 
  left_join(gen_tab, by = "ID") %>% 
  relocate(stratum, .after = last_col()) %>% 
  select(!c(ID, gender, age,age_yrs_oct2014.1, age.group)) # age and gender removed as we treat them as confounders
# sum statistic with sex and age stratum as blocking factor.
data.set.1
```


microbiome
## Type of vars  
IMPORTANT STEP 
```{r eval=FALSE}

variables_to_test <- c("chronic_cvd_2014","Roken_2014",
                       "antimicrobials_2014","AstmaCOPD_2014",
                       "statins_2014", "alpa_beta_blockers_2014")

TYPES <- c("ordinal","continuous",
           rep("ordinal",length(taxa_select)),
           rep("continuous",5),
           "categorical")

types <- c("categorical","continuous", 
           rep("categorical",length(variables_to_test)),
           rep("dirac.and.continuous",length(taxa_select)),
           rep("continuous",5),
           "categorical")


#TYPES <- c("ordinal","continuous",rep("ordinal",nocl),
 #          rep("continuous",5),
  #         "categorical")

#types <- c("categorical","continuous", 
 #          rep("categorical",length(vars.test)-2),
  #         rep("dirac.and.continuous",length(bacterial.abundance)),
   #        rep("continuous",5),
    #       "categorical")
#colnames(data.for.testing)[1:10]



```

## Test associations   

```{r eval=FALSE}
#get column number that match bacterial abundance
bac_cols <- which(colnames(data.set.1) %in% taxa_select)

no.of.factors <- length(variables_to_test)
aux.columns <- bac_cols
variables.of.interest <- names(data.set.1)
nominal.bound.on.FDR <- 0.20
log.scale <- TRUE

set.seed(1938)
B <- 100000
```


```{r eval=FALSE}
dir.create("../../test_stat")
check_association(data.set.1,
                  B = 100000,
                  no.of.factors= no.of.factors,
                  aux.columns = aux.columns,
                  log.scale = TRUE,
                  nominal.bound.on.FDR = 0.25,
                  phen.data = phenotypic.data.1,
                  compare.label="case_control",
                  aux.TYPES=TYPES,
                  cat.types = types,
                  path_loc="../../test_stat/",
                  name.of.stratification= "Sex and age-group")
```







